src/bootstrap/mod.rs

```diff
@@ -0,0 +1,88 @@
+use bindings::Windows::Win32::{
+    Foundation::HWND,
+    UI::WindowsAndMessaging::{MessageBoxW, MB_ICONERROR, MB_OK},
+};
+
+mod tests;
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub struct PackageVersion {
+    pub revision: u8,
+    pub build: u8,
+    pub minor: u8,
+    pub major: u8,
+}
+
+impl PackageVersion {
+    pub fn new(major: u8, minor: u8, build: u8, revision: u8) -> Self {
+        Self {
+            revision,
+            build,
+            minor,
+            major,
+        }
+    }
+
+    fn to_major_minor(self) -> u32 {
+        ((self.major as u32) << 8) | self.minor as u32
+    }
+}
+
+#[link(name = "Microsoft.ProjectReunion.Bootstrap")]
+extern "system" {
+    fn MddBootstrapInitialize(
+        majorMinorVersion: u32,
+        versionTag: *const u16,
+        minVersion: PackageVersion,
+    ) -> windows::HRESULT;
+
+    fn MddBootstrapShutdown() -> windows::HRESULT;
+}
+
+/// Locates the Windows App framework package compatible with the (currently internal)
+/// versioning criteria and loads it into the current process. If multiple packages meet
+/// the criteria, the best candidate is selected.
+pub fn initialize() -> windows::Result<()> {
+    initialize_without_dialog()
+    .map_err(|outer_error| {
+        unsafe {
+            MessageBoxW(
+                HWND::default(),
+                "To run this application, the Windows App preview runtime must be installed.\n\nhttps://aka.ms/projectreunion/0.8preview",
+                "This application could not be started",
+                MB_OK | MB_ICONERROR,
+            );
+            outer_error
+        }
+    })
+}
+
+pub fn initialize_without_dialog() -> windows::Result<()> {
+    let version_tag: Vec<u16> = "preview".encode_utf16().collect();
```

Comment:
Taking a dependency on https://crates.io/crates/const-utf16 or some equivalent might be helpful.

Reply:
This should soon be baked in, so perhaps we can avoid the extra dependency for the short term.

Reply:
Yep, eventually we'll pass this requirement onto the developer; they must provide the correct version of the dynamic dependency component. Still in talks with the Windows App SDK folks to figure out exactly how all this is going to work, so went with _working values_ for now.